File: frontend/src/types/interview.ts
/** Category types for interview questions */
export type InterviewCategory = "knowledge" | "behavioral" | "case-study" | "technical" | "system-design" | string;

/** Difficulty/proficiency level for questions */
export type DifficultyLevel = "beginner" | "intermediate" | "advanced";

/** A single interview question returned by the API */
export interface InterviewQuestion {
  id: string;
  question: string;
  category: InterviewCategory;
  difficulty: DifficultyLevel;
  evaluation_criteria: string[];
  model_answer: string;
}

/** A full interview session containing multiple questions */
export interface InterviewSession {
  id: string;
  created_at: string;
  questions: InterviewQuestion[];
}

/** Feedback received after submitting an answer */
export interface AnswerFeedback {
  strengths: string[];
  areas_for_improvement: string[];
  overall_assessment: string;
}

File: backend/app/schemas/interview.py
"""Pydantic models for mock interview preparation."""

from datetime import datetime
from uuid import UUID

from pydantic import BaseModel, Field

from .common import ProficiencyLevel, QuestionCategory


class InterviewQuestion(BaseModel):
    id: UUID
    question: str
    category: QuestionCategory
    difficulty: ProficiencyLevel
    model_answer: str
    evaluation_criteria: list[str] = Field(min_length=1)


class AnswerFeedback(BaseModel):
    strengths: list[str]
    areas_for_improvement: list[str]
    overall_assessment: str


class ProgressInfo(BaseModel):
    percentage: int
    completed_plans: int
    total_plans: int


class InterviewSessionResponse(BaseModel):
    """Wraps generated questions together with the session metadata."""
    id: UUID
    questions: list[InterviewQuestion]
    created_at: datetime
